src/securityPolicy.c

```diff
@@ -739,12 +773,15 @@ static inline security_policy_t _scriptWitnessPolicy(const bip44_path_t* path, b
 	}
 }
 
-static inline security_policy_t _poolRegistrationOwnerWitnessPolicy(const bip44_path_t* path)
+static inline security_policy_t _poolRegistrationOwnerWitnessPolicy(const bip44_path_t* witnessPath, const bip44_path_t* poolOwnerPath)
 {
-	switch (bip44_classifyPath(path)) {
+	switch (bip44_classifyPath(witnessPath)) {
 
 	case PATH_ORDINARY_STAKING_KEY:
-		WARN_UNLESS(bip44_isPathReasonable(path));
+		if (poolOwnerPath != NULL) {
+			DENY_UNLESS(bip44_pathsEqual(witnessPath, poolOwnerPath));
+		}
```

Comment:
```suggestion
		if (poolOwnerPath != NULL) {
			// an owner was given by path
			// the witness path must be identical
			DENY_UNLESS(bip44_pathsEqual(witnessPath, poolOwnerPath));
		} else {
			// no owner was given by path
			// we must not allow witnesses because they might witness owners given by key hash
			DENY();		
		}
```

Reply:
We also need a ledgerjs test for this case.

Reply:
Done.